### PR TITLE
LPD 27101 TECHNICAL TASK v0 SF Component annotation

### PR DESCRIPTION
**Pull request by kyle-miho**
https://liferay.atlassian.net/browse/LPD-27101

cc: @drewbrokke @nicolas-sss

@ling-alan-huang Could you review this when you have a chance? Thanks!

Updated from https://github.com/kyle-miho/liferay-portal/pull/1254

### CODE REVIEW

**Comment by kyle-miho** [comment]
ci:test:sf

**Comment by kyle-miho** [comment]
SF issue is due to an unrelated commit, should be fixed once the base branch gets updated

File: modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/BaseAddComponentAnnotationCheck.java [hunk 1]
```diff
@@ -55,7 +56,17 @@ protected abstract String getAnnotationContent(
...
-		return StringUtil.merge(lines, StringPool.NEW_LINE);
+		StringBundler sb = new StringBundler();
+
+		for (String line : lines) {
+			sb.append(line);
+
+			sb.append(StringPool.NEW_LINE);
+		}
+
+		sb.setIndex(sb.index() - 1);
```
[the remark this answers is not in the document]
Comment (kyle-miho) [reply modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/BaseAddComponentAnnotationCheck.java:67]: @drewbrokke No I agree I like your method too, we should just fix it in the loop instead, I'll fix it and resend

**Comment by kyle-miho** [comment]
@drewbrokke Fixed!

**Comment by kyle-miho** [comment]
ci:test:sf

**Review by drewbrokke — APPROVED** [review]

**Comment by kyle-miho** [comment]
ci:test:sf

**Comment by kyle-miho** [comment]
ci:test:hotel

**Comment by ling-alan-huang** [comment]
I will have a look at it later today.

**Comment by ling-alan-huang** [comment]
@kyle-miho Please see https://github.com/liferay-devtools/liferay-portal/pull/99

**Pull request closed without merging**